src/main.rs

```diff
@@ -1,51 +1,77 @@
 use log::debug;
 use std::{
-    env,
+    env, fs,
     io::{self, stdout, BufRead, Write},
 };
 use yaiwr::Calc;
 
 fn main() {
     env_logger::init();
     let args: Vec<String> = env::args().collect();
+    debug!("cli args {:?}", &args[1..]);
     if args.len() > 1 {
-        eval_print(&args[1]);
+        if args[1].ends_with(".yaiwr") {
+            run_from_file(&args[1])
+        } else {
+            eval_line(&args[1]);
+        }
     } else {
-        let stdin = io::stdin();
-        loop {
-            print!("👉 ");
-            stdout().flush().ok();
-            match stdin.lock().lines().next() {
-                Some(Ok(ref l)) => {
-                    if l.trim().is_empty() {
-                        continue;
-                    }
-                    eval_print(l);
+        repl();
+    }
+}
+
+fn run_from_file(file_name: &str) {
+    let contents = fs::read_to_string(file_name).expect("Should have been able to read the file");
```

Comment:
In a later PR, I think you should make it so that all errors bubble up to the top-level where the error is printed in one place.

Reply:
make sense, update readme todo list :point_right:  75a448ae13f57eb0fe0dd6f717a62e5e72abc49c